docs/design/architectural-overview.md

```diff
@@ -0,0 +1,173 @@
+
+## How does Kuadrant integrate with Gateway Providers
+
+The Kuadrant data plane, integrates with an Istio based gateway provider only currently: 
+
+- It registers Authorino with the `IstioOperator` as an auth provider so that Authorino can be used as an external auth provider.
+
+- It leverages an `EnvoyFilter`  to register the rate limiting service as a upstream service.
+
+- Based on the Kuadrant AuthPolicy, It leverages Istio's `AuthorizationPolicy` resource to configure when a request should trigger Authorino to be called for a given host, path and method etc. 
+
+- It provides a WebAssembly (WASM) Plugin that conforms to the [Proxy WASM ABI](https://github.com/proxy-wasm/spec) (application binary interface). This WASM Plugin is loaded into the underlying Envoy based gateway provider and configured via the Kuadrant Operator based on defined `RateLimitPolicy` resources. This binary is executed in response to a HTTP request being accepted by the gateway via the underlying Envoy instance that provides the proxy layer for the Gateway (IE Envoy). This plugin is configured with the correct upstream rate limit service name and when it sees a request, based on the provided configuration, it will trigger a call to the installed Limitador that is providing the rate limit capabilities and either allow the request to continue or trigger a response to the client with a 429 (too many requests) HTTP code.
+
+
+
+
+### Data Flows
+
+There are several different data flows when using Kuadrant. 
+
+**Control plane configuration and status reporting**
+
+The initial creation of these APIs (gateways, policies etc) is done by the relevant persona in the control plane just as they would any other k8s resource. We use the term cluster admin or gateway admin as the operations type persona configuring, and placing gateways. 
+As shown above, in a multi-cluster configuration. API definitions are pulled from the Hub and "manifested" into the spokes. The Status of those synced resources are reported back to the Hub. The same happens for a single cluster, the only difference being the work agent hub controllers are all installed on one cluster.
+
+**3rd party enforcement and Integration**
+
+In order to enforce the policy configuration, components in the control plane and data plane can reach out to configured 3rd parties such as cloud based DNS provider, TLS providers and Auth providers.
+
+
+**Request Flow**
+
+Requests coming through the gateway instance can be sent to Limitador based on configuration of the WASM plugin installed into the Envoy based gateway provider or to Authorino based configuration provided by the Istio `AuthorizationPolicy`. 
+Each of these components have the capability to see the request and need to in order to make the required decision. Each of these components can also prevent the request from reaching its intended backend destination based on user configuration. 
+
+
+![](./images/request-flow.png)
```

Comment:
I don't think we need to change anything here necessarily, but I do wonder whether the depicted connection between Authorino and the external Auth Provider in the request flow diagram could be interpreted as a requirement, when actually it is not.

Partially, it may also be a matter of interpretation about what "external" means.

The part that is true is that Kuadrant does not manage user/client identities (i.e. Kuadrant is not an IAM system). In this sense, Authorino needs to go elsewhere to fetch something that will later allow it to verify the authentication tokens supplied in the requests.

That elsewhere can be an external auth provider (e.g. to fetch a set of JSON Web Keys) or the Kuberentes cluster itself (to pull API key secrets and x509 CA certs). Either way, definitely not part of the request flow, and perhaps not "external" depending on your POV.

Exceptional cases where Authorino will perform a request to a third-party authenticator every time (caching aside), as part of the request flow, are:
- Authentication based on OAuth opaque tokens;
- Kubernetes TokenReview – this one arguably not "external".

As I said, no need to change the diagram I think. It's good that integrating a proper auth provider is somehow represented. Devil's in the details tho. It's not always external, and it's rarely something that happens in the request flow.

Reply:
I think it is a fair point. I will merge as is for now so that the overview is present and visible but in future will make sure to take this into account.